bootstrap/scripts/validation.py

```diff
@@ -0,0 +1,179 @@
+from email_validator import validate_email, EmailNotValidError
+from functools import wraps
+from pathvalidate import is_valid_filepath, ValidationError
+from shutil import which
+from typing import Callable, Optional
+from zoneinfo import available_timezones
+import CloudFlare
+import netaddr
+import re
+import requests
+import sys
+
+DISTRIBUTIONS             = ["k0s", "k3s", "talos"]
+CILIUM_LOADBALANCER_MODES = ["dsr", "snat"]
+GLOBAL_CLI_TOOLS          = ["age", "cloudflared", "flux", "sops", "jq", "kubeconform", "kustomize"]
+TALOS_CLI_TOOLS           = ["talosctl", "talhelper"]
+K0S_CLI_TOOLS             = ["k0sctl"]
+
+def required(*keys: str):
+    def wrapper_outter(func: Callable):
+        @wraps(func)
+        def wrapper(data: dict, *args, **kwargs) -> None:
+            for key in keys:
+                if data.get(key) is None:
```

Comment:
`if not data.get(key):` is better

Reply:
I get an error with that change:

```
  File "/Users/devin/Code/repos/onedr0p/flux-template-cluster/bootstrap/scripts/loader.py", line 29, in __init__
    validation.validate_acme_email(data)
  File "/Users/devin/Code/repos/onedr0p/flux-template-cluster/bootstrap/scripts/validation.py", line 25, in wrapper
    raise ValueError(f"Missing required key {key}")
ValueError: Missing required key bootstrap_acme_production_enabled
task: Failed to run task "configure": exit status 1
```

Reply:
Okay, if it breaks then go with what you have.  👍